saleor/core/middleware.py

```diff
@@ -78,3 +80,14 @@ def middleware(request):
         return get_response(request)
 
     return middleware
+
+
+def taxes(get_response):
+    """Assign tax rates for default country to `request.taxes`."""
+    def middleware(request):
+        # fixme: find out if we should or can use request.country here
+        request.taxes = get_taxes_for_country(
```

Comment:
1. Why not using `cached_property` instead of calling `get_taxes_for_country` every request so we only retrieve it if we really need it? That would save a few CPU cycles in some requests.
2. What about (maybe a settings to disable it), trying to get the tax from `request.user.default_billing_address.country` if `default_billing_address` is not blank and user is not a guest?

Reply:
The second part sounds like an interesting approach for situations where a session country is not set and GeoIP would be used as a fallback. We could instead set the session variable from the default billing address.

Reply:
I've added lazy evaluation for `request.taxes` and changed it to use `request.country` instead of always going with `settings.DEFAULT_COUNTRY`. Thanks!

As for no.2, fallbacking to `default_billing_address.country` before GeoIP seems like good material for separate issue.